rust/day02/src/main.rs

```diff
@@ -0,0 +1,73 @@
+use std::fs::read_to_string;
+
+// Clone and Copy mean this is a very small object; don't bother with pointers/references ever --
+// just copy it.
+// PartialEq lets us see if x == y
+#[derive(Clone, Copy, PartialEq)]
+enum Choice {
+    Rock,
+    Paper,
+    Scissors,
+}
+// This lets us say Rock instead of Choice::Rock
+use Choice::*;
+
+impl Choice {
+    /// Does this choice beat the other one? `None` if it's a draw
```

Comment:
`Option<bool>` is a bit of a weird way to express something with 3 outcomes - win, lose, draw. We could also make another enum. Or to have less indirection, assume self is player1 and other is player2, and return the 0, 3, 6 score